### ISSUE TITLE
`INSERT ... RETURNING` not working with C++ prepared statements

### ISSUE THREAD

**Issue by cathaypacific8747** [duckdb/duckdb#8310]
### What happens?

Using `INSERT ... RETURNING` with:
- `con.Query()`: works perfectly, data both returned and inserted.
- `con.Prepare().Execute()`: data returned but data is not inserted.

### To Reproduce

```cpp
duckdb::DuckDB db(nullptr);
duckdb::Connection con(db);
con.Query("CREATE TABLE test(val VARCHAR);");

con.Query("INSERT INTO test(val) VALUES ('query') RETURNING *")->Fetch()->Print();
con.Query("INSERT INTO test(val) VALUES (?) RETURNING *;", "query_arg")->Fetch()->Print();
con.Prepare("INSERT INTO test(val) VALUES ('prepared') RETURNING *;")->Execute()->Fetch()->Print();
con.Prepare("INSERT INTO test(val) VALUES (?) RETURNING *;")->Execute("prepared_arg")->Fetch()->Print();

con.Query("SELECT * FROM test;")->Print();
```

Output:
```yml
Chunk - [1 Columns]
- FLAT VARCHAR: 1 = [ query]

Chunk - [1 Columns]
- FLAT VARCHAR: 1 = [ query_arg]

Chunk - [1 Columns]
- FLAT VARCHAR: 1 = [ prepared]

Chunk - [1 Columns]
- FLAT VARCHAR: 1 = [ prepared_arg]

val
VARCHAR
[ Rows: 2]
query
query_arg
```

### OS:

Ubuntu 22.10, x86_64

### DuckDB Version:

0.8.1

### DuckDB Client:

C++ API

### Full Name:

Abraham Cheung

### Affiliation:

Hong Kong University of Science and Technology

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Tmonster** [comment]
Hi cathaypacific8747,

This should be fixed in 0.9.0, for whatever your use can was would you mind double checking this has indeed been fixed and close the issue?

Thanks,
Tom Ebergen

**Comment by cathaypacific8747** [comment]
I can confirm that it is indeed fixed for v0.9.0, thank you!
